modules/DNN/Perceptron.h

```diff
@@ -8,14 +8,14 @@ namespace DirectGraphicalModels {
 		class CPerceptron {
 		public:
 			DllExport CPerceptron(const std::vector<int>& vNumNeurons);
-			DllExport CPerceptron(const std::vector<ptr_nl_t>& vpLayers);
-			DllExport CPerceptron(const CPerceptron&) = delete;
+			DllExport CPerceptron(const std::vector<ptr_nl_t>& vpLayers, const std::vector<ptr_nl_t>& numHiddenLayer); // added numHiddenLayer for passing
```

Comment:
I think the addition of a new argument is not a good idea...
All the layers should be stored in `vpLayers` . We know exactly that the first element of that vector is the input layer, the last element - the output layer, and thus everything inside should be the hidden layers. At the construction step we don't even care how many of the hidden layers there are.
I think this file should not be modified at all.